kpops/utils/dict_differ.py

```diff
@@ -15,16 +17,63 @@ class DiffType(str, Enum):
     CHANGE = "change"
     REMOVE = "remove"
 
+    @staticmethod
+    def from_str(label: str) -> DiffType:
+        return DiffType[label.upper()]
+
+
+T = TypeVar("T")
 
-class Change(BaseModel):
-    old_value: Any | None
-    new_value: Any | None
 
+@dataclass
+class Change(Generic[T]):  # Generic NamedTuple requires Python 3.11+
```

Comment:
why do we need to make it generic? We only use it with a fixed type T

Reply:
`T` is used as a `TypeVar` here to avoid `Any` for the value type. I've changed it from https://github.com/bakdata/kpops/pull/88#discussion_r1094407585 so that Mypy can do its type checking.
Does that answer your question? I wasn't sure what you mean exactly

Reply:
Why don't we use unknown or dict? as far as i have seen, dict is always the tye of the generic

Reply:
`test_diff.py` uses `dict`, `int`, and `None` https://github.com/bakdata/kpops/blob/e1c7de6667baf8ef09d911dcf050c3df43695111/tests/utils/test_diff.py#L120

Reply:
based on the existing tests and your answers in https://github.com/bakdata/kpops/pull/88#discussion_r1094407585 I am expecting the values to have various possible types.

Reply:
Yeah but in the prod code it's always dict, right?

Reply:
I don't have an answer for that. The Helm diff only uses `dict`. But this class was added for the topic config diff and there I don't know the possible types. @raminqaf answered that he has seen various different types

Reply:
Pretty sure that the Helm diff also uses different types. So I am not even sure if dict is correct. There was a reason that it was `any` before

Reply:
If this requires python 3.11+ so we cannot support 3.10 anymore?

Reply:
generic dataclasses are supported on Python 3.10. only generic namedtuples (which would make sense to use) require 3.11+